marshmallow/marshalling.py

```diff
@@ -174,7 +174,7 @@ def _run_validator(self, validator_func, output,
             # Store or reraise errors
             if err.field_names:
                 field_names = err.field_names
-                field_objs = [fields_dict[each] for each in field_names]
+                field_objs = [fields_dict[each] if each in fields_dict else None for each in field_names]
```

Comment:
Ah, fair enough. I do see a use case for keeping the lists the same order and length. OK, this is fine.

Could you fix this so that it passes the flake8 test?